Open Judge System/OJS.Services/Business/Participants/ParticipantsBusinessService.cs

```diff
@@ -28,16 +28,15 @@ public class ParticipantsBusinessService : IParticipantsBusinessService
             bool isAdmin)
         {
             Participant participant;
-            if (contest.IsOnline)
+            if (contest.IsOnline && isOfficial)
             {
-                participant = new Participant(contest.Id, userId, isOfficial)
+                participant = new Participant(contest.Id, userId, isOfficial: true)
                 {
                     ParticipationStartTime = DateTime.Now,
```

Comment:
You should probably initialize the fields using a ternary operator and  save yourself the need for the else statement.